wrappers/modules/shared.nix

```diff
@@ -3,6 +3,19 @@
   options = {
     enable = lib.mkEnableOption "nixvim";
 
+    nixpkgs.useGlobalPackages = lib.mkOption {
+      type = lib.types.bool;
+      default = true; # TODO: Added 2024-12-23; switch to false one release after adding a deprecation warning
+      defaultText = lib.literalMD ''`true`, but will change to `false` in a future version.'';
+      description = ''
+        Whether Nixvim should use the host configuration's `pkgs` instance.
+        When false, `pkgs` will be constructed from `nixpkgs.source`.
+
+        > [!TIP]
+        > The host configuration is usually home-manager, nixos, or nix-darwin.
```

Comment:
Instead of a generic "tip", this could actually use a context option to dynamically produce platform-specific docs. This is possible because we actually show the per-platform options separately.

Probably out of scope for this PR though.